### PR TITLE
DBZ-2461 LSN must not be flushed after connection close

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.redhat.com/browse/DBZ-2461

### CODE REVIEW

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/DebeziumEngineIT.java [hunk 1]
```diff
@@ -197,4 +198,61 @@ public void shouldSerializeToCloudEvents() throws Exception {
...
+            }
+        }).build();
+
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        executor.execute(engine);
+
+        for (int i = 0; i < 100; i++) {
+            TestHelper.execute("INSERT INTO tests VALUES(default)");
+        }
+        Thread.sleep(5000);
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/DebeziumEngineIT.java:247]: Can we avoid this via Awaitility?

**Review by gunnarmorling — COMMENTED** [review]
LGTM, but one comment inline. Thx!

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/DebeziumEngineIT.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/DebeziumEngineIT.java:247]: Probably yes but I am not sure if it is worth the effort. This is just a way how to increase the probaility that the bug is visible, not for the test stability. TO use awaitility we would need to monitor offset store and validate if there are offsets of some condition.

Comment (gunnarmorling) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/DebeziumEngineIT.java:247]: I'm not so much concerned about stability, but about overall execution time of the test suite.

**Comment by gunnarmorling** [comment]
@jpechane test failures in PG might be related?

**Comment by gunnarmorling** [comment]
@jpechane, any thoughts on ^^^?

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/DebeziumEngineIT.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/DebeziumEngineIT.java:247]: We are not going to short it down much anyway. We need to wait for 3 secs at least to make sure the offset committer commits offsets. I don't want to reduce that interval too much as with loer values the issue tended to disappear.

Comment (gunnarmorling) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/DebeziumEngineIT.java:247]: Ok, not super-excited, but so be it.

Comment (jpechane) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/DebeziumEngineIT.java]: Removed as by-product of fixing block issue

**Comment by jpechane** [comment]
@gunnarmorling This should be stable and good for the review again.

**Comment by gunnarmorling** [comment]
Squashed and merged. Thx, @jpechane!

**Pull request merged**